### PR TITLE
Zulip Scripts to fetch user stats

### LINKED ISSUE #38: feat: Zulip stats for the user
## Description
As a User, I should be able to view my personal zulip stats on my dashboard.

## Acceptance Criteria
The code handles retrieving :

- [ ] username
- [ ] Total messages
- [ ] Add date of last time the stats were updated

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 day

### PR DESCRIPTION
**Pull request by bismitaguha**
### Description

- Added Zulip API scripts and respective models to store most recent data

Fixes #38 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: osp/migrations/0018_remove_zulipstat_last_activity.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Generated by Django 3.0.7 on 2020-08-15 00:24
```
Comment (Monal5031) [inline osp/migrations/0018_remove_zulipstat_last_activity.py]: Please optimize the migrations, I guess there are a few which just add/remove same field

File: osp/models/user_information.py [hunk 2]
```diff
@@ -9,13 +9,17 @@ class UserInformation(models.Model):
...
+        related_name='user_name',
     )
     name = models.CharField(max_length=255)
     user_type = models.CharField(
         max_length=7,
         choices=user_types.USERS,
         default='student'
     )
+    zulip_id = models.BigIntegerField(
+        blank=True
```
Comment (Monal5031) [inline osp/models/user_information.py:20]: add default as null

File: osp/views/user_information.py [hunk 3]
```diff
@@ -18,10 +19,13 @@ def get_queryset(self):
     def create(self, request):
         user = self.request.user
         serializer = UserInformationSerializer(data=request.data)
-
-        if UserInformation.objects.filter(user_id=user.id).exists():
-            return Response("Information already filled", status=status.HTTP_409_CONFLICT)
+        response = get_zulip_user(request.data['zulip_id'])
+        if response['result']=='success':
```
Comment (Monal5031) [inline osp/views/user_information.py]: pep8...

File: osp/views/user_information.py [hunk 4]
```diff
@@ -18,10 +19,13 @@ def get_queryset(self):
...
+            if UserInformation.objects.filter(user_id=user.id).exists():
```
Comment (Monal5031) [inline osp/views/user_information.py]: use this condition before executing zulip script, would save us API calls

File: osp/utils/zulip_api.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import os
+from zulip import Client
+
+from rest_framework.response import Response
+
+client = Client(config_file='~/download')
```
Comment (Monal5031) [inline osp/utils/zulip_api.py]: add this as var to env and then use it

Comment (Monal5031) [reply osp/utils/zulip_api.py]: Also document what do we need to do to create the API key and whatever we'll need during deployment

**Review by Monal5031 — CHANGES_REQUESTED** [review]

File: osp/utils/zulip_api.py (hunk 5 above)
Comment (bismitaguha) [reply osp/utils/zulip_api.py]: The file is available for download only, and variable throws error. only `config_file` is being read properly.

Comment (Monal5031) [reply osp/utils/zulip_api.py]: What I meant:
1. Document where we need to place the config file as in the exact path
2. move `'~/download'` to env and read them in settings maybe as: `ZULIP_CONFIG_PATH`

Comment (bismitaguha) [reply osp/utils/zulip_api.py]: It cannot read from a different file. Zulip API file has a different format, I opened and saw! So in comments I have added the directory the file should be kept in.

**Review by Monal5031 — APPROVED** [review]

**Pull request merged**